CoreDailies.cs

```diff
@@ -347,44 +347,63 @@ public void CollectorClass()
         }
     }
 
-    public void Cryomancer()
+public void Cryomancer()
     {
         Core.Logger("Daily: Cryomancer Class");
         if (Core.CheckInventory("Cryomancer", toInv: false))
         {
             Core.Logger("You already own Cryomancer, Skipped");
             return;
         }
-        if (!Core.IsMember && !CheckDaily(3966, true, "Glacera Ice Token"))
-            return;
-        if (Core.IsMember && !CheckDaily(3965, true, "Glacera Ice Token"))
-            return;
-        if (Core.IsMember)
+        
+        if (Core.IsMember && CheckDaily(3965, true, "Glacera Ice Token"))
+        {
+            Core.EquipClass(ClassType.Farm);
             DailyRoutine(3965, "frozentower", "Frost Invader", "Dark Ice");
-        else
+            Core.Logger($"Glacera Ice Token {Bot.Inventory.GetQuantity("Glacera Ice Token")} / 84");
+            Core.ToBank("Glacera Ice Token");
+        }
+        
+        if (CheckDaily(3966, true, "Glacera Ice Token"))
+        {
+            Core.EquipClass(ClassType.Farm);
             DailyRoutine(3966, "frozentower", "Frost Invader", "Dark Ice");
-        Core.ToBank("Glacera Ice Token");
+            Core.Logger($"Glacera Ice Token {Bot.Inventory.GetQuantity("Glacera Ice Token")} / 84");
+            Core.ToBank("Glacera Ice Token");
+        }        
+        if (Core.CheckInventory("Glacera Ice Token", 84))
+            Core.BuyItem("frozenruins", 1056, "Cryomancer", shopItemID: 3041);
+            Core.ToBank("Glacera Ice Token");
     }
 
     public void Pyromancer()
     {
         Core.Logger("Daily: Pyromancer Class");
         if (Core.CheckInventory("Pyromancer", toInv: false))
         {
-            Core.Logger("Skipped");
+            Core.Logger("You already own Pryomancer, Skipped");
             return;
         }
-        if (!Core.IsMember && !CheckDaily(2209, true, "Shurpu Blaze Token"))
-            return;
-        if (Core.IsMember && !CheckDaily(2210, true, "Shurpu Blaze Token"))
-            return;
-        Core.EquipClass(ClassType.Solo);
         Bot.Quests.UpdateQuest(2157);
-        if (Core.IsMember)
+        if (!Core.IsMember && !CheckDaily(2210, true, "Shurpu Blaze Token"))
```

Comment:
remove the !'s  this is for if the player *is* member and the daily *is* doable.